cmd/push.go

```diff
@@ -289,7 +290,7 @@ func buildImage(ctx context.Context, dev *model.Dev, imageFromApp, oktetoRegistr
 		BuildArgs:  buildArgs,
 		OutputMode: pushOpts.Progress,
 	}
-	if err := build.Run(ctx, buildOptions); err != nil {
```

Comment:
It's different because here we are applying it to a Build specification of the dev environment, on the other hand on the deploy command we were calling `build.Run` for every build specification on the full manifest